### PR TITLE
i18n(zh-cn): translate src/content/docs/zh-cn/guides/middleware.mdx

### PR DESCRIPTION
**Pull request by aaronlamz**
#### What kind of changes does this PR include?

- Translated content

#### Description

translate src/content/docs/zh-cn/guides/middleware.mdx

@liruifengv

### CODE REVIEW

File: src/content/docs/zh-cn/guides/middleware.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+这是 `onRequest()` 的可选参数，并且可能提供由中间件返回的必需的 `Response`。
+
+### `locals`
+这是一个包含在 `Response` 中的数据，可以在中间件内部操作的对象。
+
+此 `locals` 对象在请求处理过程中被传递，并作为 [`APIContext`](/zh-cn/reference/api-reference/#endpoint-context) 和 `AstroGlobal` 的属性可用。这允许在中间件，API路由，和 `.astro` 页面之间共享数据。这对于在渲染步骤中存储请求特定的数据（例如用户数据）很有用。
+
+`locals` 是一个在单个 Astro 路由中创建和销毁的对象；当你的路由页面被渲染时，`locals` 将不再存在，将会创建一个新的。需要在多个页面请求之间保持的信息必须存储在其他地方。
+
+:::提示
```
Comment (liruifengv) [inline src/content/docs/zh-cn/guides/middleware.mdx:200]:
```suggestion
:::tip
```

**Review by liruifengv — CHANGES_REQUESTED** [review]
A little change. Other LGTM!

File: src/content/docs/zh-cn/guides/middleware.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+### `onRequest()`
+这是从 `src/middleware.js` 导出的必需函数，在渲染每个页面或 API 路由之前都会调用。它接受两个可选参数：[context](#context) 和 [next()](#next)。 `onRequest()` 必须返回一个 `Response` ：可以直接返回，或者通过调用 `next()` 返回。
+
+### `context`
+这是一个包含了要在渲染过程中提供给其他中间件，API路由和`.astro` 路由的信息的对象。
+
+这是传递给 `onRequest()` 的可选参数，可能包含 [`locals`](#locals)  对象以及在渲染期间要共享的任何其他属性。例如， `context` 对象可能包含用于认证的cookies。
+
+这是传递给 API 路由的相同的 [`context`](/zh-cn/reference/api-reference/#endpoint-context)  对象。
```
Comment (Yan-Thomas) [inline src/content/docs/zh-cn/guides/middleware.mdx:185]:
```suggestion
这是传递给 API 路由的相同的 [`context`](/zh-cn/reference/api-reference/#端点上下文) 对象。
```

File: src/content/docs/zh-cn/guides/middleware.mdx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+### `next()`
+
+这是一个拦截（读取和修改）`Request`的 `Response` 或调用链中的"下一个"中间件并返回一个 `Response` 的函数。例如，这个函数可以修改响应的HTML正文。
+
+这是 `onRequest()` 的可选参数，并且可能提供由中间件返回的必需的 `Response`。
+
+### `locals`
+这是一个包含在 `Response` 中的数据，可以在中间件内部操作的对象。
+
+此 `locals` 对象在请求处理过程中被传递，并作为 [`APIContext`](/zh-cn/reference/api-reference/#endpoint-context) 和 `AstroGlobal` 的属性可用。这允许在中间件，API路由，和 `.astro` 页面之间共享数据。这对于在渲染步骤中存储请求特定的数据（例如用户数据）很有用。
```
Comment (Yan-Thomas) [inline src/content/docs/zh-cn/guides/middleware.mdx:196]:
```suggestion
此 `locals` 对象在请求处理过程中被传递，并作为 [`APIContext`](/zh-cn/reference/api-reference/#端点上下文) 和 `AstroGlobal` 的属性可用。这允许在中间件，API路由，和 `.astro` 页面之间共享数据。这对于在渲染步骤中存储请求特定的数据（例如用户数据）很有用。
```

File: src/content/docs/zh-cn/guides/middleware.mdx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+:::
+
+### `sequence()`
+
+这是一个接受中间件函数作为参数，并按照传递的顺序执行它们的函数。
+
+
+### `createContext`
+
+一个用于创建 [`APIContext`](/zh-cn/reference/api-reference/#endpoint-context) 的底层 API，Astro 中间件可以使用它。
```
Comment (Yan-Thomas) [inline src/content/docs/zh-cn/guides/middleware.mdx:211]:
```suggestion
一个用于创建 [`APIContext`](/zh-cn/reference/api-reference/#端点上下文) 的底层 API，Astro 中间件可以使用它。
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks for the great work, and welcome to Team Docs @aaronlamz!

**Pull request merged**